src/main/java/org/embulk/output/sftp/SftpFileOutput.java

```diff
@@ -211,27 +202,68 @@ public TaskReport commit()
 
     private void closeCurrentFile()
     {
-        if (currentFile == null) {
-            return;
+        if (localOutput != null) {
+            try {
+                localOutput.close();
+            }
+            catch (IOException ex) {
+                throw Throwables.propagate(ex);
+            }
         }
+    }
 
+    private Void uploadFile(final String remotePath)
+    {
         try {
-            currentFileOutputStream.close();
-        }
-        catch (IOException e) {
-            logger.info(e.getMessage());
+            return retryExecutor()
+                    .withRetryLimit(maxConnectionRetry)
+                    .withInitialRetryWait(500)
+                    .withMaxRetryWait(30 * 1000)
+                    .runInterruptible(new Retryable<Void>() {
+                        @Override
+                        public Void call() throws InterruptedException, IOException, RetryGiveupException
+                        {
+                            FileObject remoteFile = newSftpFile(getSftpFileUri(remotePath));
+                            logger.info("new sftp file: {}", remoteFile.getPublicURIString());
+                            try (BufferedOutputStream outputStream = new BufferedOutputStream(remoteFile.getContent().getOutputStream())) {
+                                try (BufferedInputStream inputStream = new BufferedInputStream(new FileInputStream(tempFile))) {
+                                    IOUtils.copy(inputStream, outputStream);
```

Comment:
I'm not sure but, we don't need to delete existing remote files before the copy process is started?